server/src/sockets/chess.socket.js

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable no-param-reassign */
+import IO from 'socket.io';
+
+import { assignOpponentSocket, Games } from './socket.utils';
+
+const games = new Games();
+
+const chessSocket = (server) => {
+  const io = IO(server);
+  io.on('connection', (socket) => {
+    const game = games.findGame(socket.id);
+    socket.game = game;
+    if (game.player2) {
+      const opponent = assignOpponentSocket(socket.game, socket.id);
+      io.to(opponent.socketId).emit('gameData', { game });
+    }
```

Comment:
What happens if player2 is not found?
Both in client and server?
is the client blocked, until it does?

Reply:
If player 2 is not found the server will send the game object to the client with the status of the game as idle. On the client side, we can show a loading screen until another player connects and sets the game status to live.